WordPress/Classes/Services/EditorSettingsService.swift

```diff
@@ -5,8 +5,20 @@ import Foundation
 }
 
 @objc class EditorSettingsService: LocalCoreDataService {
-    @objc func syncEditorSettings(for blog: Blog, success: @escaping () -> Void, failure: @escaping (Swift.Error) -> Void) {
-        guard let api = blog.wordPressComRestApi() else {
+    let wpcomApi: WordPressComRestApi?
+
+    @objc convenience override init(managedObjectContext: NSManagedObjectContext) {
+        self.init(managedObjectContext: managedObjectContext, wpcomApi: nil)
+    }
+
+    init(managedObjectContext: NSManagedObjectContext, wpcomApi: WordPressComRestApi? = nil) {
```

Comment:
I thought passing an API here that overrides the specific blog one for testing was a bit unexpected. I looked around and what I see other services doing is have a `remoteForBlog:` method that gets overriden by the tests. Something like c19ae6ef4fa749328edcf9da121c70d2f87697f1

Reply:
Nice trick! Absolutely :shipit:

Reply:
Cherry-picked the commit for easier integration with this PR